### PR TITLE
feat: Nearest places on the map (BE)

### PR DESCRIPTION
**Pull request by AverchenkoAA**
https://trello.com/c/BU1y2uVd/17-feat-nearest-places-on-the-map-be

### CODE REVIEW

**Comment by AverchenkoAA** [comment]
[(feat) Nearest places on the map (BE)](https://trello.com/c/BU1y2uVd/17-feat-nearest-places-on-the-map-be)

File: apps/api/src/app/services/sights/sights.service.ts [hunk 1]
```diff
@@ -52,9 +48,16 @@ export class SightsService {
...
       .doc(id)
       .delete();
-    if (!snapshot) {
-      return Promise.resolve(false);
-    }
-    return Promise.resolve(true);
+    return Promise.resolve(!!snapshot);
+  }
+
+  public async getSightByCoordinates(coordinate: Coordinates): Promise<Sight> {
```
Comment (BaturevichNikita) [inline apps/api/src/app/services/sights/sights.service.ts]: Try to take out the duplicate code in a separate method. I mean getSightByCoordinates and getSightByID

File: apps/api/src/app/authorization/auth.middleware.ts [hunk 2]
```diff
@@ -58,11 +59,11 @@ export class AuthorizationMiddleware {
   }
 
   private getUserID(authServ: AuthService) {
-    return async (error: Error, response: express.Response, body: any) => {
+    return async (error: Error, response: express.Response, body) => {
       if (!error && response.statusCode == 200) {
-        let user = JSON.parse(body);
-        let currentUser: User[] = await authServ.getUsersByID(user.id);
-        if (!currentUser[0]) {
+        const user = JSON.parse(body);
```
Comment (BaturevichNikita) [inline apps/api/src/app/authorization/auth.middleware.ts]: Use === instead ==

File: apps/api/src/app/services/geolib/geo.service.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  // or returns all sights using points array to get sights from DB.
+  private async getSomeSightsFromPoints(
+    points: Coordinates[],
+    amount: number
+  ): Promise<Sight[]> {
+    if (amount > points.length || !amount) {
+      amount = points.length;
+    }
+
+    const allights: Sight[] = await this.sightsService.getAllSights();
```
Comment (BaturevichNikita) [inline apps/api/src/app/services/geolib/geo.service.ts]: Typo `allights`.
Have you tried something like `.where('sight.coordinates', 'in', points)`,
where `points = [{latitude: '123', longitude: '456'}]`

File: apps/api/src/app/services/geolib/geo.service.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
+import { Injectable, Inject } from '@mogilev-guide/api/ioc';
+import { Sight, Coordinates } from '@mogilev-guide/models';
+import { SightsService } from '@mogilev-guide/api/services/sights';
+import * as geolib from 'geolib';
+
+@Injectable()
+export class GeoService {
+  @Inject() private sightsService!: SightsService;
+
+  public async getOneSightsFromPoints(startPoint: Coordinates): Promise<Sight> {
```
Comment (BaturevichNikita) [inline apps/api/src/app/services/geolib/geo.service.ts]: getOneSightsFromPoints => getSightFromPoint

File: apps/api/src/app/services/geolib/geo.service.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    const nearestPoint = geolib.findNearest(startPoint, allPoints);
+
+    const nearestSight: Sight = await this.sightsService.getSightByCoordinates(
+      nearestPoint as Coordinates
+    );
+
+    return nearestSight;
+  }
+
+  public async getAmountSightsFromPoints(
```
Comment (BaturevichNikita) [inline apps/api/src/app/services/geolib/geo.service.ts]: getAmountSightsFromPoints => getSightsFromPoint

**Pull request closed without merging**